orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -61,13 +61,80 @@ NodeArg* GetDimsValue(Graph& graph, NodeArg* input, NodeArg* indices_arg, Node&
   return gather_out_args[0];
 }
 
+// Insert Expand to the in_index-th input of node.
+// The node should have two inputs and the shape of the other input (node.InputDefs()[1-in_index]) should be
+// [batch_size, seq_len, ...]. This function insert an Expand to expand shape of the in_index-th input of node with
+// a shape arg of [batch_size, seq_len, 1, 1, ...] which size is equal with node.InputDefs()[1-in_index]->Shape().size.
+NodeArg* InsertExpandForNodeInput(Graph& graph,
+                                  Node& node,
+                                  uint32_t in_index,
+                                  NodeArg* first_two_dims_arg,
+                                  const logging::Logger& logger) {
+  auto input_shape = node.InputDefs()[1 - in_index]->Shape();
+  NodeArg* expand_shape_arg = nullptr;
+  if (input_shape->dim_size() <= 2) {
```

Comment:
is it possible input_shape->dim_size() < 2?

Reply:
I think input_shape->dim_size() can only be 2 or 3, right? If so, can we add a check?

Reply:
done , thx